src/Microsoft.ML.AutoML/ColumnInference/ColumnPurpose.cs

```diff
@@ -13,6 +13,7 @@ internal enum ColumnPurpose
         TextFeature = 4,
         Weight = 5,
         ImagePath = 6,
-        SamplingKey = 7
+        SamplingKey = 7,
+        LabelFeature,   // CategoricalFeature that requires ValueToKey converter, better naming?
```

Comment:
I think we'd want to use `HashToKey` (name be off) instead of the mentioned `ValueToKey` as the `ValueToKey `will map future unseen values to NA in your test dataset; and as a lesser issue is slow by taking a full pass of the dataset.